### PR TITLE
feat(TTD): Filter task based on completion and priority

### PR DESCRIPTION
**Pull request by mohammed-sharraf-sazim**
Description:
- Added filtering option to filter tasks based on priority (High, Medium and Low)
- Added filtering option to filter tasks based on completion (Active/Completed)

Associated Ticket Link(s):
-https://trello.com/c/MFh6HH9N/9-filter-tasks-active-completed
-https://trello.com/c/vIYNQtnE/10-filter-tasks-priority

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -32,6 +35,23 @@ function App() {
...
 
+  const filteredTasks = tasks.filter(task => {
+    if (filter === 'all' && priorityFilter === 'all') {
+      return true;
+    } else if (filter === 'completed' && priorityFilter === 'all') {
+      return task.isCompleted;
+    } else if (filter === 'active' && priorityFilter === 'all') {
+      return !task.isCompleted;
+    } else if (filter === 'all' && priorityFilter !== 'all') {
+      return task.priority === priorityFilter as Priority; // Cast to Priority
```
Comment (mMohsinSazim) [inline src/App.tsx:46]: first remove the comment, also see how to implement this without using casting

**Review by mMohsinSazim — CHANGES_REQUESTED** [review]

File: src/components/FilterTasks.tsx [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+  SelectLabel,
+  SelectTrigger,
+  SelectValue,
+} from "../components/ui/select";
+import { Priority } from "../Model";
+
+interface Props {
+  filter: string;
+  setFilter: React.Dispatch<React.SetStateAction<string>>;
+  priorityFilter: string | Priority;
```
Comment (nur-zaman-sazim) [inline src/components/FilterTasks.tsx:15]: `Priority`  is enough. No need to to use string.
If you need any more constants add it in the enum.

File: src/components/FilterTasks.tsx [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+        </SelectTrigger>
+        <SelectContent>
+          <SelectGroup>
+            <SelectLabel>State of Task</SelectLabel>
+            <div className="flex flex-col space-y-2">
+              <button
+                className={`bg-dark-gray-100 px-4 py-2 rounded-md ${
+                  priorityFilter === "all" ? "bg-blue-500 text-white" : "text-gray-700"
+                }`}
+                onClick={() => setPriorityFilter("all")}
```
Comment (nur-zaman-sazim) [inline src/components/FilterTasks.tsx:74]: Use an enum for priority filter.

**Review by nur-zaman-sazim — CHANGES_REQUESTED** [review]

File: src/App.tsx (hunk 1 above)
Comment (mohammed-sharraf-sazim) [reply src/App.tsx]: Removed comments and implemented without casting

File: src/components/FilterTasks.tsx (hunk 3 above)
Comment (mohammed-sharraf-sazim) [reply src/components/FilterTasks.tsx:74]: If I use enum I have to make some major changes.

**Review by nur-zaman-sazim — APPROVED** [review]

**Pull request merged**